cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/SsaInternals.qll

```diff
@@ -108,6 +108,101 @@ predicate hasRawIndirectInstruction(Instruction instr, int indirectionIndex) {
   )
 }
 
+private module IteratorDefUse {
+  private import semmle.code.cpp.models.interfaces.Iterator as Interfaces
+  private import semmle.code.cpp.models.implementations.Iterator as Iterator
+  private import semmle.code.cpp.models.implementations.StdContainer as StdContainer
+
+  private CallInstruction getCall(ArgumentOperand operand) { result.getAnOperand() = operand }
+
+  /**
+   * Gets a `CallInstruction` that targets an iterator-returning function that was
+   * used to construct the value used by `use`.
+   */
+  private CallInstruction getAnIteratorAccess(UseImpl use) {
+    exists(Instruction value, StoreInstruction store |
+      // Get the ultimiate definition of this use.
+      store = use.getPruningUse().getAnUltimateValue().asInstruction() and
+      value = store.getSourceValue()
+    |
+      // We're done if it's a call that targets an iterator-returning function.
+      if value.(CallInstruction).getStaticCallTarget() instanceof Iterator::GetIteratorFunction
+      then result = value
+      else
+        // Otherwise, we recursively search for a prior definition.
+        exists(Operand operand |
+          // Recurse through identity conversions
+          conversionFlow(operand, value, _)
+          or
+          // ... and dereferences.
+          exists(CallInstruction call |
+            isDereference(call, operand) and
+            operandForfullyConvertedCall(store.getDestinationAddressOperand(), call)
+          )
+        |
+          result = getAnIteratorAccess(any(UseImpl priorUse | priorUse.getOperand() = operand))
+        )
+    )
+  }
+
+  /**
+   * Holds if `containerUse` is the use of a variable of a container type that is used to obtain
+   * an iterator into the container represented by `containerUse`, and `iteratorUse` is a use
+   * of this iterator.
+   */
+  private predicate acquiresIteratorFromContainer(DirectUse containerUse, DirectUse iteratorUse) {
+    exists(CallInstruction getIterator |
+      containerUse.getOperand() = getIterator.getThisArgumentOperand() and
+      containerUse.getIndirectionIndex() = 1 and
+      getIterator = getAnIteratorAccess(iteratorUse)
+    )
+  }
+
+  /**
+   * Holds if `use` is a use of an iterator that is obtained by calling an iterator-returning
+   * function that returns an iterator to the container represented by `containerUse.`
```

Comment:
`isIteratorDef` has the final period after the quote.

Reply:
Fixed in 3144fe92e238cad6c916068406c60c077ac365c3.